### PR TITLE
TDAP058-Mark-Done-Functionality-Using-Redux

### PR DESCRIPTION
**Pull request by amit-ib**
Mark Done Functionality Using Redux

### CODE REVIEW

File: src/redux/todoReducer.tsx [hunk 1]
```diff
@@ -17,23 +17,35 @@ const todoReducer = (state: todoList = initialState, action: any) => {
...
+        tasks: remainingTasks,
+      };
+
+    case "MARK_DONE_TODO":
+      let todoList = [...state.tasks];
+      todoList.forEach((item) => {
+        if (item.id === action.payload.id) {
+          item.isDone = !item.isDone;
+        }
+      });
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/redux/todoReducer.tsx]: Updated and used findIndex

File: src/redux/todoActions.tsx [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
+  console.log("deleteTodo", data);
+  return {
+    type: "DELETE_TODO", // returned type property
+    payload: data,
+  };
+};
+
+export const markDoneTodo = (data: TodoModal) => {
+  return {
+    type: "MARK_DONE_TODO",
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/redux/todoActions.tsx]: Done

File: src/redux/todoReducer.tsx [hunk 3]
```diff
@@ -17,23 +17,35 @@ const todoReducer = (state: todoList = initialState, action: any) => {
       if (action.payload) {
         updatedTodos.push(action.payload);
       }
-     
+
       return {
         ...state, // copy origenal state and update only required
-       tasks:updatedTodos
+        tasks: updatedTodos,
```
[the remark this answers is not in the document]
Comment (amit-ib) [reply src/redux/todoReducer.tsx:27]: Noted

**Review by akash-dhande-ib — APPROVED** [review]
Minor corrections, rest looks good.

**Pull request merged**